client/src/Pages/Food/index.js

```diff
@@ -1,10 +1,101 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import IconLabeledButton from '../../components/Button';
+import ContainerComponent from '../../components/Container';
+import TextInputField from '../../components/InputFeild';
+import CardComponent from '../../components/Card';
+
+const axios = require('axios');
 
 function FoodPage() {
+  const [allFood, setAllFood] = useState({});
+  const [errorMessage, setErrorMessage] = useState('');
+  const [searchValue, setSearchValue] = useState('');
+  const [foodFilter, setFoodFilter] = useState('');
+  useEffect(() => {
+    const { CancelToken } = axios;
+    const source = CancelToken.source();
+    (async () => {
+      try {
+        const food = await axios.get('/api/v1/food');
+        return setAllFood(food);
+      } catch (err) {
+        return setErrorMessage(
+          err.response.data.message || 'something went wrong'
+        );
+      }
+    })();
+    return () => source.cancel('Operation canceled by the user.');
+  });
```

Comment:
This effect will run on every component render - in other words, every time the component's state changes.  It probably doesn't need to run that often (since it is just re-fetching the same food list each time) and that could be a performance problem (because it is calling to the server every time the user types a letter in the search field, for example).
The fix for this is easy, just pass an empty array as the second argument to useEffect:
```
useEffect(() => {
    const { CancelToken } = axios;
    const source = CancelToken.source();
    (async () => {
      ....
      ....
   })();
    return () => source.cancel('Operation canceled by the user.');
  }, []); //Notice the empty array here.
```
That second argument is called the dependency array and controls when the effect runs.  If you pass an empty array like that, it will only run once ever, when the component mounts (which is what we want here, I think).  If you don't pass that argument at all, the effect will re-run on every render (every prop or state change).  If you pass named variables in the array (i.e. `[searchValue]`) it will only re-run when those specific variables change.

Some more info here:
https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

Reply:
I only want this to run once to get me all the data from the database and they will be saved in the allFood variable.
so I will be adding an empty array. however, I still don't fully understand. in this case for authentication, I have to check the status if it's 401 or 200, so if I understand correctly, this will be run every time this page is called which means it will always check the status once at the beginning and store all the data in allFood?

Reply:
Yes I think that’s correct. It will run this useEffect only when this component “mounts”, which means when it’s added to the page. The only way it would run again is if this component is then removed from the screen (for example the user navigates to a different page inside the app) and then returns to this food page (thus mounting a new instance of this component). 
Does that help clear up the confusion?

If you need a way to re-fetch the food data after the component mounts because the user decides to log in (so, user goes from unauthenticated to authenticated) you could do that by passing their auth status as a prop to this component, then adding that prop to this useEffects dependencies array - that’d cause the effect to re-run when the auth prop changes c